src/test/cli-options.test.ts

```diff
@@ -0,0 +1,249 @@
+          name: 'main',
+        },
+        extraArgs: ['--extra'],
+      });
+    })
+  );
+
+  // Does not work in pnpm, see https://github.com/pnpm/pnpm/issues/4821.
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  (agent === 'pnpm' ? test.skip : test)(
+    `${agent} test -- --extra`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} test -- --extra`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'test',
+        },
+        extraArgs: ['--extra'],
+      });
+    })
+  );
+
+  test(
+    `${agent} start -- --extra`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} start -- --extra`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'start',
+        },
+        extraArgs: ['--extra'],
+      });
+    })
+  );
+
+  test(
+    `${agent} run main --watch`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} run main --watch`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'main',
+        },
+        watch: true,
+      });
+    })
+  );
+
+  // Does not work in pnpm, see https://github.com/pnpm/pnpm/issues/4821.
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  (agent === 'pnpm' ? test.skip : test)(
+    `${agent} test --watch`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} test --watch`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'test',
+        },
+        watch: true,
+      });
+    })
+  );
+
+  test(
+    `${agent} start --watch`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} start --watch`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'start',
+        },
+        watch: true,
+      });
+    })
+  );
+
+  test(
+    `${agent} run main --watch -- --extra`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} run main --watch -- --extra`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'main',
+        },
+        extraArgs: ['--extra'],
+        watch: true,
+      });
+    })
+  );
+
+  // Does not work in pnpm, see https://github.com/pnpm/pnpm/issues/4821.
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  (agent === 'pnpm' ? test.skip : test)(
+    `${agent} test --watch -- --extra`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} test --watch -- --extra`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'test',
+        },
+        extraArgs: ['--extra'],
+        watch: true,
+      });
+    })
+  );
+
+  test(
+    `${agent} start --watch -- --extra`,
+    timeout(async ({rig}) => {
+      await assertOptions(rig, `${agent} start --watch -- --extra`, {
+        script: {
+          packageDir: rig.temp,
+          name: 'start',
+        },
+        extraArgs: ['--extra'],
+        watch: true,
+      });
+    })
+  );
```

Comment:
I think for a couple versions we should detect someone doing using `watch` instead of `--watch` and give them a special error message, just because this change is going to catch people (and their existing scripts and workflows)

Something like:

```
As of wireit v0.6, use `--watch` instead of `watch`. This was added to support passing flags to scripts on the command line.

This will be an error for a couple of versions, and after that the `watch` argument will be passed to the underlying script.
```

Reply:
I second this.

Reply:
Done